src/main/java/com/mathworks/ci/RunMatlabTestsStep.java

```diff
@@ -0,0 +1,161 @@
+
+    @DataBoundConstructor
+    public RunMatlabTestsStep() {
+        
+    }
+    
+    public String getTestResultsTAP() {
+        return testResultsTAP;
+    }
+
+    @DataBoundSetter
+    public void setTestResultsTAP(String testResultsTAP) {
+        this.testResultsTAP = testResultsTAP;
+    }
+    
+    public String getTestResultsPdf() {
+        return testResultsPdf;
+    }
+
+    @DataBoundSetter
+    public void setTestResultsPdf(String testResultsPdf) {
+        this.testResultsPdf = testResultsPdf;
+    }
+
+    public String getTestResultsJUnit() {
+        return testResultsJUnit;
+    }
+
+    @DataBoundSetter
+    public void setTestResultsJUnit(String testResultsJUnit) {
+        this.testResultsJUnit = testResultsJUnit;
+    }
+
+    public String getCodeCoverageCobertura() {
+        return codeCoverageCobertura;
+    }
+
+    @DataBoundSetter
+    public void setCodeCoverageCobertura(String codeCoverageCobertura) {
+        this.codeCoverageCobertura = codeCoverageCobertura;
+    }
+
+    public String getTestResultsSimulinkTest() {
+        return testResultsSimulinkTest;
+    }
+
+    @DataBoundSetter
+    public void setTestResultsSimulinkTest(String testResultsSimulinkTest) {
+        this.testResultsSimulinkTest = testResultsSimulinkTest;
+    }
+
+    public String getModelCoverageCobertura() {
+        return modelCoverageCobertura;
+    }
+    
+
+    @DataBoundSetter
+    public void setModelCoverageCobertura(String modelCoverageCobertura) {
+        this.modelCoverageCobertura = modelCoverageCobertura;
+    }
+
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+       
+        return new MatlabStepExecution(context,constructCommandForTest(getInputArgs()), COPY_SCRATCH_FILE);
+    }
+    
+    @Extension
+    public static class CommandStepDescriptor extends StepDescriptor {
+
+        @Override
+        public Set<? extends Class<?>> getRequiredContext() {
+            return ImmutableSet.of(TaskListener.class, FilePath.class, Launcher.class,
+                    EnvVars.class, Run.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return Message.getValue("matlab.tests.build.step.name");
+        }
+    }
+    
+    public String constructCommandForTest(String inputArguments) {
+        final String matlabFunctionName =
+                FilenameUtils.removeExtension(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE);
+        final String runCommand = "exit(" + matlabFunctionName + "(" + inputArguments + "))";
+        return runCommand;
+    }
+
+    
+    private String getInputArgs() {
+        final List<String> inputArgs = new ArrayList<>();
+        final Map<String, String> args = getMatlabArgs();
+
+        args.forEach((key, val) -> {
+            if (val != null) {
+                inputArgs.add("'" + key + "'" + "," + "'" + val + "'");
+            }
+        });
```

Comment:
What if someone has single quotes in their specified filename? I believe this will break. I think you need to escape single quotes in values.

Reply:
in Script user can embed this in single quotes like below as it fails to parse .
report: 'filen'm.xml' 

Now user can use double quotes instead.

report:"filen'm.xml"  

above work fine through java and it send proper parameter to MATLAB file however MATLAB fails as you suspected.

![image](https://user-images.githubusercontent.com/47204011/82541362-f7d6a000-9b6d-11ea-9055-9b035f318e01.png)

I think we should escape them in .m file instead in java  what is your thought

Reply:
Done